lib/kapnismology/application_information.rb

```diff
@@ -1,22 +1,47 @@
+require "json"
+
 module Kapnismology
   # This class provides information about the running environment the smoketest is being
   # executed under.
   class ApplicationInformation
-    INFO_UNKNOWN = ''
+    GIT_SHOW_COMMAND = "git show HEAD --abbrev-commit --oneline".freeze
+    ECS_CONTAINER_METADATA_FILE = ENV["ECS_CONTAINER_METADATA_FILE"].freeze
+    INFO_UNKNOWN = "".freeze
 
     def trace_id
       Object.const_defined?(:Trace) ? Trace.id.trace_id.to_s : INFO_UNKNOWN
     end
 
     def codebase_revision
-      latest_commit_info.split(/\s/).first || INFO_UNKNOWN
+      ref = latest_commit_info
+      ref ? ref.split(/\s/).first : INFO_UNKNOWN
     rescue Errno::ENOENT, StandardError
-     INFO_UNKNOWN
+      INFO_UNKNOWN
     end
 
     private
+
     def latest_commit_info
-      `git show HEAD --abbrev-commit --oneline`
+      latest_ref_from_git || latest_ref_from_ecs_metadata || nil
+    end
+
+    def latest_ref_from_git
+      result = `#{GIT_SHOW_COMMAND}`
+      $?.success? ? result : nil
+    end
+
+    # This assumes the ImageName features the git ref as the last part of a string delimited by dots.
+    # See spec/support/ecs_metadata.json as well as
+    # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html
+    def latest_ref_from_ecs_metadata
+      return if ECS_CONTAINER_METADATA_FILE.nil?
+      return unless !ECS_CONTAINER_METADATA_FILE.strip.empty? && File.readable?(ECS_CONTAINER_METADATA_FILE)
+
+      begin
+        JSON.parse(File.read(ECS_CONTAINER_METADATA_FILE))["ImageName"].split(".").last[0...7]
```

Comment:
sorry, that's what happens when I comment late at night. I also meant to say do `&.` or in other way make this not blow up if the format is not exactly as expected. Just return nil. Maybe even log if the information can't be found

Reply:
This kind of errors are caught in the calling method that has a "catch all" rescue.
Thinking of which I guess it doesn't make much sense to do that begin/rescue here if we're not going to do anything with that specific error.